Tests/Provider/YouTubeProviderTest.php

```diff
@@ -206,4 +206,11 @@ public function testHelperProperties()
         $this->assertSame(100, $properties['player_parameters']['height']);
         $this->assertSame(100, $properties['player_parameters']['width']);
     }
+
+    public function testGetReferenceUrl()
+    {
+        $media = new Media();
+        $media->setProviderReference('123456');
+        $this->assertEquals('http://www.youtube.com/watch?v=123456', $this->getProvider()->getReferenceUrl($media));
+    }
```

Comment:
This is getting repetitive… maybe you could introduce an abstract test case with a `getExpectedUrl($providerReference)` method ?